csrc/scheduler/normalization.cpp

```diff
@@ -1787,15 +1788,17 @@ void schedulePersistentKernelInnerOuter(
       inner_reduction_tvs,
       cached_inputs,
       cached_outputs,
-      boundaryNodesSet);
+      selected_tvs_inner);
   // Propagate outer reduction. Each outer reduction is connected with its
   // cached_gmem and output, since we added all the cached_gmem to the
   // boundaryNodesSet, the transformation from one outer reduction can't
   // propagate to other outer reductions due to the cutoff at boundaryNodesSet.
   // Thus, we need a loop to initiate the propagation from each outer reduction.
-  // Parallelization will not propagate to inner reductions as they are
-  // transformed differently.
+  // Don't allow parallelization propagation goes through cached_gmem, see issue
+  // 246.
   for (long unsigned int i = 0; i < outer_reference_tvs.size(); i++) {
+    const auto& selected_tvs_outer =
+        scheduler_utils::getAllTvsFrom(outer_reduction_tvs, {cached_gmem[i]});
```

Comment:
Shouldn't this be `outer_reduction_tvs[i]`?

Reply:
You are right!